File: Notepad_bavarna/src/application/utils/BaseBillingWithOffer.java
package application.utils;

public abstract class BaseBillingWithOffer extends BaseBilling{
	
	protected double discountPercentage;

	public void applyDiscount(double percentage) {
		this.discountPercentage = percentage;
	}
	
	public double calculateBill(double cost, double tax) {
		return cost + (cost * tax ) - (cost * discountPercentage);
	}

}

File: Aparna/Notepad/Billing/BaseBillingWithOffer.cs


namespace Notepad.Billing
{
    abstract class BaseBillingWithOffer : Billing
    {
        protected double discountPercentage { get; set; }
        public abstract void applyDiscount(double discountPercentage);

        public override double CalculateBill(double cost, double tax)
        {
            return cost + (cost * tax) - (cost * discountPercentage);
        }
    }
}
